### PR TITLE
Drop v1 contracts packages everywhere

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Version 1 of the Symfony contracts have been released during the Symfony 4.x days, but still many Symfony 6 components declare compatibility with them:

https://github.com/symfony/symfony/blob/a8bbf632df06b26bbd8a62f22203054bbff34d32/src/Symfony/Component/DependencyInjection/composer.json#L22

This circumstance complicates our build matrix and makes issues like #48792 hard to track down.

Version 2 was released together with Symfony 5 which means that we really only need v1 contracts if we want to actively support Symfony 4 components which are EOL and already unsupported by Symfony 6. I make the bold assumption that no Symfony 6 app should be required to pin any of the contracts packages to v1 or even an outdated v2 minor release.

This is why I propose to drop support for contracts v1 everywhere it did not happen already, making `^2.5 || ^3` the loosest constraint we use for a contracts package on a Symfony 6.3 component or bundle.

### CODE REVIEW

File: src/Symfony/Bridge/ProxyManager/composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
         "php": ">=8.1",
         "friendsofphp/proxy-manager-lts": "^1.0.2",
         "symfony/dependency-injection": "^6.3",
-        "symfony/deprecation-contracts": "^2.1|^3"
+        "symfony/deprecation-contracts": "^2.5|^3"
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/ProxyManager/composer.json:22]: why bump deprecation-contracts? better not if not needed don't you think?

Comment (derrabus) [reply src/Symfony/Bridge/ProxyManager/composer.json:22]: For consistency, mainly. 2.5 is the release we still maintain in our LTS branch. Do you want to define a different target 2.x minor version for each contracts package? If so, let's define them and I adjust the PR.

Comment (nicolas-grekas) [reply src/Symfony/Bridge/ProxyManager/composer.json:22]: OK.
I looked for bumping to v3 but that'd make our packages less compatible with third party packages.

**Review by nicolas-grekas — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @derrabus.

**Pull request merged**